src/dbnode/storage/block/merged_block_reader.go

```diff
@@ -129,8 +129,9 @@ func (r *dbMergedBlockReader) mergedReader() (xio.BlockReader, error) {
 		r.readers[i] = nil
 	}
 
+	stream, _ := r.encoder.Stream()
```

Comment:
Shouldn't perhaps `ok` be checked here?

Reply:
The original code just passed in nil and as far as I can tell the xio.BlockReader handles nil fine

```golang
// CloneBlock returns a clone of the block with the underlying data reset
func (b BlockReader) CloneBlock(pool pool.CheckedBytesPool) (BlockReader, error) {
	sr, err := b.SegmentReader.Clone(pool)
	if err != nil {
		return EmptyBlockReader, err
	}
	return BlockReader{
		SegmentReader: sr,
		Start:         b.Start,
		BlockSize:     b.BlockSize,
	}, nil
}

// IsEmpty returns true for the empty block
func (b BlockReader) IsEmpty() bool {
	return b.SegmentReader == nil && b.Start.Equal(timeZero) && b.BlockSize == 0
}

// IsNotEmpty returns false for the empty block
func (b BlockReader) IsNotEmpty() bool {
	return !b.IsEmpty()
}

// ResetWindowed resets the underlying reader window, as well as start time and blockSize for the block
func (b *BlockReader) ResetWindowed(segment ts.Segment, start time.Time, blockSize time.Duration) {
	b.Reset(segment)
	b.Start = start
	b.BlockSize = blockSize
}
```

It looks like I'll need to update the xio.BlockReader to detect empty segments though since we want to change Stream() to return an empty segment instead of nil

Reply:
actually I think for this implementation of the block reader it doesnt matter